binding/Binding/HandleDictionary.cs

```diff
@@ -45,6 +53,16 @@ internal static TSkiaObject GetOrAddObject<TSkiaObject> (IntPtr handle, bool own
 			if (handle == IntPtr.Zero)
 				return null;
 
+			if (SkipObjectRegistrationType.IsAssignableFrom (typeof (TSkiaObject))) {
+#if THROW_OBJECT_EXCEPTIONS
+				throw new InvalidOperationException (
+					$"For some reason, the object was constructed using a factory function instead of the constructor. " +
+					$"H: {handle.ToString ("x")} Type: {typeof (TSkiaObject)}");
+#else
+				return objectFactory.Invoke (handle, owns);
+#endif
+			}
```

Comment:
Technically "slow" but should never be hit because the constructors are being used directly. But, just to be safe we include the code here. However, debug runs will actually throw as well.